jupyter_core/application.py

```diff
@@ -115,14 +117,19 @@ def config_files(self):
         if self.config_file:
             return [self.config_file]
         else:
-            return []
+            return [self.config_file_name]
```

Comment:
If this always returns a list of 1 item, should we rename it and just have it return that item? Or do subclasses override it to return lists of other lengths?